drivers/vmwarevsphere/vsphere.go

```diff
@@ -316,20 +316,23 @@ func (d *Driver) Create() error {
 
 	switch d.CreationType {
 	case "legacy":
-		log.Infof("Creating VM...")
+		log.Infof("creating VM...")
 		b2dutils := mcnutils.NewB2dUtils(d.StorePath)
-		if err := b2dutils.CopyIsoToMachineDir(d.Boot2DockerURL, d.MachineName); err != nil {
-			return err
+		if b2dutils != nil {
+			if err := b2dutils.CopyIsoToMachineDir(d.Boot2DockerURL, d.MachineName); err != nil {
+				return err
+			}
+			return d.createLegacy()
 		}
-		return d.createLegacy()
+		return fmt.Errorf("unable to perform legacy provisioning of VM %s", d.MachineName)
```

Comment:
Can we include why in this error message?

Reply:
removed this as there would never been a nil pointer returned for b2dutils